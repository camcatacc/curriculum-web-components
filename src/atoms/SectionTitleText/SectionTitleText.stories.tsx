import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import SectionTitleText from "atoms/SectionTitleText/SectionTitleText";

const Template: ComponentStory<typeof SectionTitleText> = (args) => <SectionTitleText {...args} />;
export const Standard = Template.bind({});
Standard.args = {
	children: "This is a Section Title"
};

export default {
	title: "Atoms/SectionTitleText",
	component: SectionTitleText
} as ComponentMeta<typeof SectionTitleText>;
