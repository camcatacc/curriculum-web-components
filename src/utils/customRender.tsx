import { ThemeProvider } from "@material-ui/styles";
import { render, RenderOptions, RenderResult } from "@testing-library/react";
import { ReactElement } from "react";
import { getMaterialTheme } from "styles/materialUi";

export const customRender = (ui: ReactElement, renderOptions?: RenderOptions & { darkMode?: boolean }): RenderResult => {
	const Wrapper: React.FC = ({ children }) => {
		return <ThemeProvider theme={getMaterialTheme(renderOptions?.darkMode ? "dark" : "light")}>{children}</ThemeProvider>;
	};

	return render(ui, { wrapper: Wrapper, ...renderOptions });
};
