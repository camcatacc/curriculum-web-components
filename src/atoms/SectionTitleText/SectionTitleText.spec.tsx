// Modules
import React from "react";
import { customRender } from "utils/customRender";

// Accessibility
import { axe, toHaveNoViolations } from "jest-axe";

// Elements
import SectionTitleText from "atoms/SectionTitleText/SectionTitleText";

// Definitions
import type { SectionTitleTextProps } from "atoms/SectionTitleText/SectionTitleText";

// Default Props
const defaultProps: SectionTitleTextProps = {
	children: "Bu"
};

// Tests
expect.extend(toHaveNoViolations);
describe("SectionTitleText", () => {
	it("Renders as expected and has no accessibility violations", async () => {
		const { container } = customRender(<SectionTitleText {...defaultProps} />);
		expect(container).toMatchSnapshot();
		expect(await axe(container)).toHaveNoViolations();
	});
	it("Renders as expected with animation", () => {
		const { container } = customRender(<SectionTitleText {...defaultProps} animated={true} />);
		expect(container).toMatchSnapshot();
	});
});
