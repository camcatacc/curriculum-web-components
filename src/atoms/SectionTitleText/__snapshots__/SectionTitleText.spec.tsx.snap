// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionTitleText Renders as expected and has no accessibility violations 1`] = `
<div>
  <div>
    <h2
      class="MuiTypography-root uppercase MuiTypography-h2 MuiTypography-alignCenter"
    >
      Bu
    </h2>
  </div>
</div>
`;

exports[`SectionTitleText Renders as expected with animation 1`] = `
<div>
  <div
    style="opacity: 0; transform: translateX(-50px) translateY(0px) translateZ(0);"
  >
    <h2
      class="MuiTypography-root uppercase MuiTypography-h2 MuiTypography-alignCenter"
    >
      Bu
    </h2>
  </div>
</div>
`;
